float64 position_x
float64 position_y
float64 position_z
float64 vector_x
float64 vector_y
float64 vector_z
---

